cpp_linter/rest_api/github_api.py

```diff
@@ -522,7 +656,138 @@ def _dismiss_stale_reviews(self, url: str):
+            url=f"{self.api_url}/graphql",
+            method="POST",
+            data=json.dumps({"query": query, "variables": variables}),
+            strict=False,
+        )
+        if response.status_code != 200:
+            logger.error(
+                "Could not get existing review thread comments: %d",
+                response.status_code,
+            )
+            return
+        data = response.json()
+        found_threads = []
+        nodes = data["data"]["repository"]["pullRequest"]["reviewThreads"]["nodes"]
+        for thread in nodes:
+            for comment in thread["comments"]["nodes"]:
+                if (
+                    comment["id"]
+                    and (
+                        not no_dismissed
+                        or (
+                            thread["isResolved"] is False
+                            and thread["isCollapsed"] is False
+                        )
+                    )
+                    and comment["body"].startswith(COMMENT_MARKER)
+                ):
+                    found_threads.append(thread)
+                    break
+        return found_threads
+
+    def _close_review_comment(
+        self, thread_id: str, comment_id: str, delete: bool = True
+    ):
+        """Resolve or Delete an existing review thread comment.
+
+        :param thread_id: Thread ID for the conversation to close (only used when ``delete``==`False`).
+        :param comment_id: The comment ID of the comment within the requested thread to close (only used when ``delete``==`True`).
+        :param delete: `True` to delete the review comment, `False` to set it as resolved.
+        """
+        mutation = RESOLVE_REVIEW_COMMENT
+        variables = {
+            "threadId": thread_id,
+        }
+        operation = "resolve"
+        if delete:
+            mutation = DELETE_REVIEW_COMMENT
+            variables = {
+                "id": comment_id,
+            }
+            operation = "delete"
+        response = self.api_request(
+            url=f"{self.api_url}/graphql",
+            method="POST",
+            data=json.dumps({"query": mutation, "variables": variables}),
+            strict=False,
+        )
+        if response.status_code != 200:
+            logger.error(
+                "Failed to %s review thread comment: %d",
+                operation,
+                response.status_code,
+            )
+        elif "errors" in response.json():
+            error_msg = response.json()["errors"][0]["message"]
+            if "Resource not accessible by integration" in error_msg:
+                logger.error(
+                    "Changing review thread comments requires `contents: write` permission."
+                )
+            else:
+                logger.error(
+                    "Failed to %s review thread comment: %s", operation, error_msg
+                )
+        else:
+            logger.debug("Review comment thread %sd: %s", operation, thread_id)
```

Comment:
`self.api_request()` comes equipped with error logging. So, if we just want to add context to the logged error, we can just say:
```suggestion
        logger.debug(
            "%s review comment thread %s (thread_id: %s)",
            operation.title(),
            "failed" if response.status_code != 200 else "succeeded",
            thread_id,
        )
```